### DISCUSSION TITLE
Easy integration, synergistic with existing syntax, to increase usability and adoption

### DISCUSSION THREAD

**Discussion by NerdyDeedsLLC** [mermaid-js/mermaid#3951] [Ideas]
## Hey, sooooo....
```
box [?[DID YOU KNOW | There's currently no way to make one of these widely-recognized (from books and sites 
worldwide) call-out boxes on GitHub (short of using graphics, like I'm doing here). This is problematic, 
though, since not all GitHub markdown fields SUPPORT images.]]
```
![image](https://user-images.githubusercontent.com/43561032/210096749-017d7a2e-3ce5-409f-808e-7c30a49c3f5a.png)

```
box [*[SUGGESTIONS/IDEAS | I'd argue that something like these (which share great synergies with the rest of 
your syntax, but are easier to type out) would find even greater use and use-cases than diagrams, charts, & 
graphs.]]
```
![image](https://user-images.githubusercontent.com/43561032/210096819-56eb43e5-6e1a-4c0a-8fec-86a90378aa28.png)

```
box [i[INFORMATIVE NOTES | The box title could simply be the string before the pipe, the body after it, and 
the ability to \| escape one allow for its use therein ]]
```
![image](https://user-images.githubusercontent.com/43561032/210097008-12f53d7a-6099-4277-b68e-1f944f77d0db.png)

```
box [i[Omitting the pipe entirely could simply cause the box to act as a space for text to be written, but 
far more attention-grabbing ]]
```
![image](https://user-images.githubusercontent.com/43561032/210097062-93b3f5e5-c328-45d0-93c2-10cb2012ffe0.png)

```
box [![Important MESSAGES | The character(s) between the square brackets could dictate the symbol 
loaded into the decor slot as well as dictating the color and style of the box.]]
```
![image](https://user-images.githubusercontent.com/43561032/210097157-5a37f2ed-5f65-46d4-a7c0-08a89eb49d6f.png)

```
box [!![VITAL NOTIFICATIONS|Simply increasing the number of bangs would be an easy way to 
select intensity...]]
```
![image](https://user-images.githubusercontent.com/43561032/210097250-ca2ef8e6-094d-4300-9d0e-ec5bb6bfe066.png)

```
box [!!![CRITICAL ALERTS|...perhaps capping out at 3?]]
```
![image](https://user-images.githubusercontent.com/43561032/210097293-caa747ba-085f-4492-bab3-d039a3d5404c.png)

----
### Please don't misunderstand me here...
I'm absolutely in love with the Mermaid functionality. But it's hard to argue that there is a colossal amount of DOCUMENTATION currently being hosted on GitHub. As of June 2022, GitHub reported having over **83 million developers and more than 200 million repositories**, including **at least 28 million public** repositories.

I think it would be reasonably safe to assert, therefore, we're looking at _220 MILLION_ `README.md` files alone. And _that's_ notwithstanding the amount of comments, formal product documentation, wikis, issues, code review, helpfiles, gists, and discussions (like this one!). All that having been said, I would be utterly unsurprised to discover fully a _billion_ pieces of text reference material that is in dire need of a mechanism to call out the critical parts.

```mermaid
flowchart LR
    id1{{ Now personally... I've long-since begun abusing them as a call-out mechanism already... }}
    style id1 fill:#fC6,stroke:#333,stroke-width:2px
```
...but it strikes me that instead of an obtuse, roundabout means of users invariably manipulating the syntax to accomplish this goal, might there not be...

```mermaid
%%{init: { "sequence": { "wrap": true, "width":300 } } }%%
sequenceDiagram
A BETTER->WAY FORWARD: That might get utilized even MORE?
WAY FORWARD->A BETTER: (even in your OWN documentation?)
A BETTER->WAY FORWARD: Just sayin'....
```

**Comment by NerdyDeedsLLC** [comment]
(I figured I'd bring it up before attempting to work out a way to _really_ abuse the functionality to generate something similar myself and bundling the result into a Chrome Extension for ease of implementation)

**Reply by ZenAtWork** [reply]
Wait, so, I assume the imagery/color/style and whatnot, are preset elements of the inserted template, and would just be automatically inserted when the box is generated by the Mermaid renderer? I suppose you could use a syntax similar to:

```
box [#BADA55|https://user-images.githubusercontent.com/123456789/123456789-.jpg[HEADER TEXT| Some message]]
```
...if you wanted to enable users to customize the color and icon, too!

**Comment by ZenAtWork** [comment]
Now THIS is a great idea. I would totally use this. I suspect my org would too. We use GH for MOST of our documentation. It'd be really nice to be able to make it more manual-esque!
